go/private/context.bzl

```diff
@@ -837,3 +838,15 @@ go_config = rule(
 
 def _expand_opts(go, attribute_name, opts):
     return [go._ctx.expand_make_variables(attribute_name, opt, {}) for opt in opts]
+
+# Used to get attribute values which may have been transitioned.
+# Transitioned attributes end up as lists.
+# We never use split-transitions, so we always expect exactly one element in those lists.
+# But if the attribute wasn't transitioned, it won't be a list.
+def _flatten(maybe_list):
```

Comment:
I would prefer a less generic name, but have a hard time finding a good one. What do you think of `_flatten_possibly_transitioned_attr`?